velox/core/QueryConfig.h

```diff
@@ -80,6 +80,11 @@ class QueryConfig {
   // truncating the decimal part instead of rounding.
   static constexpr const char* kCastToIntByTruncate = "cast_to_int_by_truncate";
 
+  // This flags allows the cast from string to date accept patterns other than
+  // the "(+-)[YYYY-MM-DD]" format.
+  static constexpr const char* kCastStringToDateNonStandard =
```

Comment:
nit: can you also mention this newly added query config in the PR description that enables this behavior.